module/zfs/metaslab.c

```diff
@@ -4789,6 +4789,34 @@ metaslab_trace_fini(zio_alloc_list_t *zal)
  * ==========================================================================
  */
 
+void
+metaslab_force_alloc(metaslab_t *msp, uint64_t start, uint64_t size,
+    dmu_tx_t *tx)
+{
+	ASSERT(msp->ms_disabled);
+	ASSERT(MUTEX_HELD(&msp->ms_lock));
+	uint64_t txg = dmu_tx_get_txg(tx);
+
+	for (uint64_t off = start; off < start + size; ) {
```

Comment:
Since `off` is being updated at the bottom of the loop I'd switch this to a `while` loop which feels more natural for this kind of thing.